### PR TITLE
[Enchancement](function) refact and optimize some function register

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

1. refact function create helper
2. use AggregateFunctionNullUnaryInline to optimize some functions:
```sql
select group_bit_or(k1),group_bit_or(k2),group_bit_or(k3),group_bit_or(k4) from d_table;
+--------------------+--------------------+--------------------+--------------------+
| group_bit_or(`k1`) | group_bit_or(`k2`) | group_bit_or(`k3`) | group_bit_or(`k4`) |
+--------------------+--------------------+--------------------+--------------------+
|                 -1 |                 -1 |                 -1 |                 -1 |
+--------------------+--------------------+--------------------+--------------------+
0.5s->0.3s
```

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.77 seconds
 stream load tsv:          471 seconds loaded 74807831229 Bytes, about 151 MB/s
 stream load json:         38 seconds loaded 2358488459 Bytes, about 59 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          27 seconds loaded 861443392 Bytes, about 30 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230220140949_clickbench_pr_100395.html

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/aggregate_functions/helpers.h [hunk 1]
```diff
@@ -48,150 +55,145 @@ namespace doris::vectorized {
...
-#define DISPATCH(TYPE)                \
-    if (which.idx == TypeIndex::TYPE) \
-        return new AggregateFunctionTemplate<TYPE, Data<TYPE>>(std::forward<TArgs>(args)...);
-    FOR_NUMERIC_TYPES(DISPATCH)
-#undef DISPATCH
-    return nullptr;
-}
-
+template <template <typename> class AggregateFunctionTemplate, typename Type>
+struct BuilerDirect {
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/helpers.h:59]: Builder

File: be/src/vec/aggregate_functions/helpers.h [hunk 2]
```diff
@@ -48,150 +55,145 @@ namespace doris::vectorized {
...
+    }
+};
+
+using creator_with_integer_type = creator_with_type_base<true, false, false>;
+using creator_with_numeric_type = creator_with_type_base<true, true, false>;
+using creator_with_decimal_type = creator_with_type_base<false, false, true>;
+using creator_with_type = creator_with_type_base<true, true, true>;
+
+template <int define_index = 0>
+struct creator_with_no_type {
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/helpers.h:166]: not_number_type

File: be/src/vec/aggregate_functions/helpers.h [hunk 3]
```diff
@@ -48,150 +55,145 @@ namespace doris::vectorized {
...
+template <bool allow_integer, bool allow_float, bool allow_decimal, int define_index = 0>
+struct creator_with_type_base {
+    template <typename Class, typename... TArgs>
+    static IAggregateFunction* create_base(const bool result_is_nullable,
+                                           const DataTypes& argument_types, TArgs&&... args) {
+        WhichDataType which(remove_nullable(argument_types[define_index]));
+#define DISPATCH(TYPE)                                                                            \
+    if (which.idx == TypeIndex::TYPE) {                                                           \
+        using T = typename Class::template Builder<TYPE>::T;                                      \
+        if (argument_types[define_index]->is_nullable()) {                                        \
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/helpers.h:99]: need check all types

Comment (HappenLee) [inline be/src/vec/aggregate_functions/helpers.h:99]: need check all types

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run clickbench

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**